core/jobs/batch_jobs/story_node_jobs.py

```diff
@@ -0,0 +1,210 @@
+                                        story_metadata.created_on))
+
+                                if (cmd['cmd'] == 'add_story_node' and
+                                    cmd['node_id'] == node['id']):
+                                    node_created_on = (
+                                        utils.get_time_in_millisecs(
+                                        story_metadata.created_on))
+                                    break
+                            if node_created_on is not None:
+                                break
+
+                        if node_created_on is None:
+                            raise Exception(
+                                'Node was not created.'
+                            )
+
+                        node_published_on = story_published_on if (
+                            story_published_on is not None and
+                            node_created_on is not None and
+                            story_published_on > node_created_on) else (
+                                node_created_on)
+                        node['first_publication_date_msecs'] = (
+                            node_published_on if node['status'] == 'Published'
+                            else None)
+                        node['planned_publication_date_msecs'] = (
+                            node['first_publication_date_msecs'])
+                        if node.get('last_modified_msecs') is None:
+                            node['last_modified_msecs'] = node_published_on
+
+                    updated_story_model_list.append(story_model)
+
+            except Exception as e:
+                logging.exception(e)
+                return result.Err((story_model.id, e))
+        return result.Ok(updated_story_model_list)
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+
+        fetched_story_models = (
+            self.pipeline
+            | 'Get story models' >> ndb_io.GetModels(
+                story_models.StoryModel.get_all())
+            | 'Add corresponding topic Ids as keys' >> beam.WithKeys(  # pylint: disable=no-value-for-parameter
+                lambda story_model: story_model.corresponding_topic_id)
+        )
+
+        fetched_topic_models = (
+            self.pipeline
+            | 'Get topic models' >> ndb_io.GetModels(
+                topic_models.TopicModel.get_all())
+            | 'Add topic Ids as keys' >> beam.WithKeys(  # pylint: disable=no-value-for-parameter
+                lambda topic_model: topic_model.id)
+        )
+
+        topic_story_pairs = (
+            (fetched_topic_models, fetched_story_models)
+            | 'Merge topic and story models' >> beam.CoGroupByKey()
+            | 'Get rid of topic ID' >> beam.Values() # pylint: disable=no-value-for-parameter
+        )
+
+        updated_story_models_results = (
+            topic_story_pairs
+            | 'Update story node fields' >> beam.Map(self._update_story_node)
+        )
```

Comment:
@Kej-r03. You are not filtering erroneous results after running `_update_story_node`. We would like to commit nothing in case any errors are encountered. There is a transform defined in `core/jobs/transforms/results_transforms.py` which does this and is used in existing jobs. PTAL and update here accordingly. Here is the story migration job reference:

https://github.com/oppia/oppia/blob/6d78839b605f3b009b93f1a41ecb66292ddb3b6b/core/jobs/batch_jobs/story_migration_jobs.py#L188

Reply:
Done